scripts/validate_docstrings.py

```diff
@@ -497,6 +497,44 @@ def parameter_desc(self, param):
                 desc = desc[: desc.index(full_directive)]
         return desc
 
+    def parameter_desc_list(self, param):
```

Comment:
I'm still not sure about this function in general. Sorry if this was discussed before but why do we need to iterate by lines to validate this? Can't we just use a multiline regex that detects if the proper punctuation exists?

Reply:
Hi @WillAyd ,
I have tried to implement this. I have removed `parameter_desc_list` and directly applied a regex on what is returned by `parameter_desc`.

I may overcomplicating this in my head, but I thought that iterating by lines and using the 'raw' parameter list as returned by the NumpyDocString class would have been a bit stricter check.